ICSharpCode.Decompiler.Tests/TestCases/Correctness/Switch.cs

```diff
@@ -37,6 +37,7 @@ public static void Main()
 			SwitchWithGoto(2);
 			SwitchWithGoto(3);
 			SwitchWithGoto(4);
+			SwitchWithEnum(5);
```

Comment:
You added the test case to the correctness tests. These tests are for checking semantic equivalence and are executed. 

This test case should be added to the pretty tests: https://github.com/icsharpcode/ILSpy/blob/master/ICSharpCode.Decompiler.Tests/TestCases/Pretty/Switch.cs

There the generated code is compared.

Reply:
For the runtime, enum values are numeric constants anyway, so the behavior of the code is the same.

Reply:
During adding a new test to [Pretty/Switch.cs](https://github.com/icsharpcode/ILSpy/blob/master/ICSharpCode.Decompiler.Tests/TestCases/Pretty/Switch.cs), I found out that my solution worked only when there is a PDB file next to the DLL. It happens because I use `HasGeneratedName` property inside my check. I will try to find a working solution.